test/functional/api/cas/ioclass_config.py

```diff
@@ -3,24 +3,154 @@
 # SPDX-License-Identifier: BSD-3-Clause-Clear
 #
 
+import enum
+import re
+import string
 from datetime import timedelta
+from random import randint, randrange
+
+from packaging import version
 
 from core.test_run import TestRun
+from test_tools import fs_utils
+from test_utils import os_utils
+from test_utils.generator import random_string
 
 default_config_file_path = "/tmp/opencas_ioclass.conf"
 
 MAX_IO_CLASS_ID = 32
-
+MAX_IO_CLASS_PRIORITY = 255
 MAX_CLASSIFICATION_DELAY = timedelta(seconds=6)
+IO_CLASS_CONFIG_HEADER = "IO class id,IO class name,Eviction priority,Allocation"
+
+
+class IoClass:
+    def __init__(self, class_id: int, rule: str = '', priority: int = None,
+                 allocation: bool = True):
+        self.id = class_id
+        self.rule = rule
+        self.priority = priority
+        self.allocation = allocation
+
+    def __str__(self):
+        return (f'{self.id},{self.rule},{"" if self.priority is None else self.priority}'
+                f',{int(self.allocation)}')
+
+    def __eq__(self, other):
+        return type(other) is IoClass and self.id == other.id and self.rule == other.rule \
+               and self.priority == other.priority and self.allocation == other.allocation
+
+    @staticmethod
+    def from_string(ioclass_str: str):
+        parts = [part.strip() for part in re.split('[,|]', ioclass_str.replace('║', ''))]
+        return IoClass(
+            class_id=int(parts[0]),
+            rule=parts[1],
+            priority=int(parts[2]),
+            allocation=parts[3] in ['1', 'YES'])
+
+    @staticmethod
+    def list_to_csv(ioclass_list: [], add_default_rule: bool = True):
+        list_copy = ioclass_list[:]
+        if add_default_rule and not len([c for c in list_copy if c.id == 0]):
+            list_copy.insert(0, IoClass.default())
+        list_copy.insert(0, IO_CLASS_CONFIG_HEADER)
+        return '\n'.join(str(c) for c in list_copy)
+
+    @staticmethod
+    def csv_to_list(csv: str):
+        ioclass_list = []
+        for line in csv.splitlines():
+            if line.strip() == IO_CLASS_CONFIG_HEADER:
+                continue
+            ioclass_list.append(IoClass.from_string(line))
+        return ioclass_list
```

Comment:
I'm not so sure if it would simplify things - I would still need to handle adding default IoClass in list_to_csv or skipping header and converting data to IoClass in csv_to_list. Additionally I feel like using csv.writer with io.StringIO would be overkill for such mundane task